drivers/pwm/pwm_stm32.c

```diff
@@ -112,6 +137,63 @@ static void (*const set_timer_compare[TIMER_MAX_CH])(TIM_TypeDef *,
 #endif
 };
 
+/** Channel to capture get function mapping. */
+#if defined(CONFIG_SOC_SERIES_STM32F0X) ||	\
+	defined(CONFIG_SOC_SERIES_STM32G0X) ||	\
+	defined(CONFIG_SOC_SERIES_STM32G0X) ||	\
+	defined(CONFIG_SOC_SERIES_STM32H5X) ||	\
+	defined(CONFIG_SOC_SERIES_STM32H7X) ||	\
+	defined(CONFIG_SOC_SERIES_STM32L4X) ||	\
+	defined(CONFIG_SOC_SERIES_STM32L5X) ||	\
+	defined(CONFIG_SOC_SERIES_STM32U5X) ||	\
+	defined(CONFIG_SOC_SERIES_STM32WBX) ||	\
+	defined(CONFIG_SOC_SERIES_STM32WLX)
+static uint32_t __maybe_unused (*const get_channel_capture[])(const TIM_TypeDef *) = {
+#else
+static uint32_t __maybe_unused (*const get_channel_capture[])(TIM_TypeDef *) = {
```

Comment:
Nit, non blocking for readability, would it be possible to use an intermediate definition and put this differentiation in a single place in this driver ?
I know this kind of code already appears in multiple places in this drivers, but raising the point while I'm looking at it..

Reply:
What do you mean using an intermediate definition? To differentiate between the `const TIM_TypeDef *` and `TIM_TypeDef *` in one place? IMO that would make it less readable, due to how inconsistent the LL drivers are. I would also prefer to keep it as is, because we've already spent a lot of time on this driver.